resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java

```diff
@@ -440,7 +440,12 @@ public void insertInterceptorPrecedenceBefore(String before, String newPrecedenc
 
    public static void pushContextDataMap(Map<Class<?>, Object> map)
    {
-      contextualData.setLast(map);
+      contextualData.push(map);

```

Comment:
It looks like AbstractAsynchronousResponse is the only place that calls that method.